src/wbi/legacy/flash_finder.py

```diff
@@ -40,11 +40,19 @@ def flash_finder(input_folder, output_folder, chunksize=4000):
             frames.astype(np.float64), axis=1)
         # f.seek(chunkMaxSize*1024*512*2*i) #not needed, np.fromfile moves the pointer
 
-    # Last partial chunk
-    chunk = np.fromfile(f, dtype=np.uint16, count=remainingframes * 1024 * 512)
-    frames = chunk.reshape((remainingframes, 1024 * 512))
-    brightness[-remainingframes - 1:-1] = np.average(frames, axis=1)
-    stdev[-remainingframes - 1:-1] = np.std(frames.astype(np.float64), axis=1)
+
+    # In cases where the number of frames in .dat file aren't exact multiple of 6 * chunk_size
+    if remainingframes > 0:
+        # Last partial chunk
+        chunk = np.fromfile(f, dtype=np.uint16, count=remainingframes * 1024 * 512)
+        frames = chunk.reshape((remainingframes, 1024 * 512))
+        try:
+            brightness[-remainingframes - 1:-1] = np.average(frames, axis=1)
+            stdev[-remainingframes - 1:-1] = np.std(frames.astype(np.float64), axis=1)
+        except ValueError:
```

Comment:
where in lines 50/51 do we expect a `ValueError` to be raised?

Reply:
Both of those lines will cause `ValueError` in cases where `chunksize // 6 ` is greater than number of frames in .dat file (in this case is 90). In both of these lines the shape of the slice from the either `brightness` or `stdev` array does not match the shape of the array resulting from `np.average(frames, axis=1)` or `np.std(frames, axis=1)`.